aten/src/ATen/native/cuda/CuFFTPlanCache.h

```diff
@@ -109,153 +258,76 @@ class CuFFTConfig {
     // For half, base strides on the real part of real-to-complex and
     // complex-to-real transforms are not supported. Since our output is always
     // contiguous, only need to check real-to-complex case.
-    if (input.scalar_type() == ScalarType::Half) {
+    if (dtype == ScalarType::Half) {
       // cuFFT on half requires compute capability of at least SM_53
       auto dev_prop = at::cuda::getCurrentDeviceProperties();
       TORCH_CHECK(dev_prop->major >= 5 && !(dev_prop->major == 5 && dev_prop->minor < 3),
                "cuFFT doesn't support signals of half type with compute "
                "capability less than SM_53, but the device containing input half "
                "tensor only has SM_", dev_prop->major, dev_prop->minor);
       for (int64_t i = 0; i < signal_ndim; i++) {
-        auto signal_size = checked_signal_sizes[i];
-        TORCH_CHECK(is_pow_of_two(signal_size),
-                 "cuFFT doesn't support signals of half type with size at any ",
-                 "dimension that is not a power of two, but got a signal size of ",
-                 checked_signal_sizes);
+        TORCH_CHECK(is_pow_of_two(sizes[i + 1]),
+            "cuFFT doesn't support signals of half type with size at any ",
```

Comment:
I suggest inverting this warning for clarity:

"cuFFT only supports dimensions whose sizes are powers of two when computing in half precision, but got..."